compiler/modules/types/function.bal

```diff
@@ -41,8 +41,7 @@ function functionSubtypeIsEmpty(Context cx, SubtypeData t) returns boolean {
         if res == () {
             // we've got a loop
             io:println("got a function loop");
-            // XXX is this right???
```

Comment:
This is right: I have checked with Giuseppe.

Reply:
I am starting to wonder whether this fix might be the right one, given we have cells. I will ask.